state-chain/custom-rpc/src/lib.rs

```diff
@@ -1484,6 +1537,34 @@ where
 			results_for_each_asset.map(|inner| inner.into_iter().flatten().collect())
 		}
 	}
+
+	fn cf_boost_pool_pending_fees(
+		&self,
+		asset: Option<Asset>,
+		at: Option<state_chain_runtime::Hash>,
+	) -> RpcResult<BoostPoolFeesResponse> {
+		let get_boost_fees_for_asset = |asset| {
```

Comment:
we use an identical pattern for the boost pool details, we could factor this out

Reply:
Are you talking about the mechanism that executes a lambda for all assets if asset is `None`? I factored out `execute_for_all_or_one_asset` now.

If you are talking about the fact that both of them call `cf_boost_pool_details`, then I don't think it is as important. Note that the lambda passed to `map` is different, and whatever code is repeated isn't worth factoring out further IMO.

Reply:
Yeah the `execute_for_all_or_one_asset` 👍